build

```
@@ -100,12 +100,30 @@ function needsSucrase(source, dest, path = '') {
 	return false;
 }
 
+function clearPath(path, src, isJs = false) {
+	try {
+		const stats = fs.lstatSync(path);
+		if (stats.isDirectory()) {
+			for (const file of fs.readdirSync(path)) {
+				const source = src + '/' + file;
+				clearPath(path + '/' + file, source, fs.existsSync(source));
+			}
+			return;
+		}
+		if (!isJs) src = src.replace('.js', '.ts');
+		if (!fs.existsSync(src)) {
+			fs.unlinkSync(path);
+		}
+	} catch (e) {}
+}
+
 function sucrase(src, out, opts) {
 	try {
 		if (!force && src !== './config' && !needsSucrase(src, out)) {
 			return false;
 		}
 	} catch (e) {}
+	clearPath(out, src); // empty the dir to get rid of built js files with no ts couterpart
```

As of Node.js 12.10.0, fs.rmdirSync supports a `recursive` option, so you can use:

```suggestion
	fs.rmdirSync(out, { recursive: true });
```

---

I thought about this, but it presents problems with things like tools/, where built stuff is in the same dir as non-built stuff. Additionally, not sure if windows supports this (we had trouble before) so I thought it might be wiser to be a bit more fine-tuned with it.

---

yes, `FS#rmdir` only really works on UNIX-like systems. https://github.com/smogon/pokemon-showdown/pull/7187#issuecomment-671115093

---

Ah, fair enough.